backend/agents/persona_agent.py

```
@@ -162,56 +484,193 @@ def _get_system_prompt(self) -> str:
+                            "persona_id": str(self.persona.id),
+                            "context_type": "conversation",
+                            "message_type": "assistant",
+                            "user_progress_id": str(user_progress_id),
+                            "scene_id": str(scene_id),
+                            "timestamp": str(datetime.now()),
+                            "session_id": self.persona_session_id  # Add session isolation
+                        }]
+                    )
+                except Exception as e:
+                    print(f"Error storing conversation in vectorstore: {e}")
+                    raise e
```

Ruff (0.14.3)

Use `raise` without specifying exception name

Remove exception name

(TRY201)